domain/application/state/types.go

```diff
@@ -595,6 +595,22 @@ type charmLocator struct {
 	ArchitectureID sql.NullInt64 `db:"architecture_id"`
 }
 
+// resourceToAdd represents a resource we want to add into the state
+type resourceToAdd struct {
+	UUID string `db:"uuid"`
+
+	ApplicationUUID string `db:"application_uuid"`
+	CharmUUID       string `db:"charm_uuid"`
+	Name            string `db:"charm_resource_name"`
+
+	Revision     *int      `db:"revision"`
+	Origin       string    `db:"origin_type_name"`
+	OriginTypeID string    `db:"origin_type_id"`
+	State        string    `db:"state_name"`
+	StateID      int       `db:"state_id"`
+	CreatedAt    time.Time `db:"created_at"`
+}
```

Comment:
todo: make resourceToAdd a bit more simple per conversation of structures doing many things at once - but not requiring all values.

```suggestion
type resourceToAdd struct {
	UUID string `db:"uuid"`
	ApplicationUUID string `db:"application_uuid"`
	CharmUUID       string `db:"charm_uuid"`
	Name            string `db:"charm_resource_name"`
	Revision     *int      `db:"revision"`
	OriginTypeID string    `db:"origin_type_id"`
	StateID      int       `db:"state_id"`
	CreatedAt    time.Time `db:"created_at"`
}

type origin struct {
	Origin       string    `db:"origin_type_name"`
	OriginTypeID string    `db:"origin_type_id"`
}

type state struct {
	State        string    `db:"state_name"`
	StateID      int       `db:"state_id"`
}
```

Reply:
Not sure it would actually be better, but I can give it a try. For me resourceToAdd do exactly what it is meant for: adding a resource.

My mental model  on such usecase revolve around some kind of input/enrich/insert pipeline, where resourceToAdd is the main data. 

So it is not an issue to build a partially defined structure => link it in some prep statement (solving ids), then insert it.

It would be an issue if it was exported data (which could mess up a potential callers).

I will push the suggestion in a separated commit.

Reply:
Code deleted (will be done in another pr)